src/components/ProjectCard/ProjectCard.js

```diff
@@ -16,31 +18,64 @@ export class ProjectCard extends Component {
 
   returnElements = () => {
     let matchingPalettes = this.findProjectPalette();
-    return matchingPalettes.map((palette) => {
+    return matchingPalettes.map((palette, i) => {
       const { name, color1, color2, color3, color4, color5 } = palette;
         return (
           <div key={uuid()} className="mapped-palettes">
-            <p className="palette-name">{name}</p>
-            <p className="color-minibox">color1: {color1}</p>
-            <p className="color-minibox">color2: {color2}</p>
-            <p className="color-minibox">color3: {color3}</p>
-            <p className="color-minibox">color4: {color4}</p>
-            <p className="color-minibox">color5: {color5}</p>
+            <p className="palette-name" id={this.createPaletteId(i)} >{name}</p>
+            <div className="color-minibox" id={this.createPaletteId(i)} style={{backgroundColor: `#${color1}`}}> </div>
```

Comment:
I like how you made the key dynamic by using the optional index param in map

Reply:
Thank you!!! 💅